docs/topics/data-analysis/data-analysis-connect-to-db.md

```diff
@@ -1,20 +1,158 @@
+1. Download and install the latest version of [IntelliJ IDEA Ultimate](https://www.jetbrains.com/idea/download/?section=mac).
+2. Install the [Kotlin Notebook plugin](https://plugins.jetbrains.com/plugin/16340-kotlin-notebook) in IntelliJ IDEA.
+
+   > Alternatively, access the Kotlin Notebook plugin from **Settings/Preferences** | **Plugins** | **Marketplace** within IntelliJ IDEA.
+   >
+   {type="tip"}
+
+3. Create a new Kotlin Notebook by selecting **File** | **New** | **Kotlin Notebook**.
+4. In the Kotlin Notebook, import the Kotlin DataFrame library by running the following command:
+
+```kotlin
+%use dataframe
+```
+
+5. Ensure you have access to an SQL database, either locally hosted, such as MariaDB or MySQL, or remotely hosted on a 
+server or cloud platform.
+
+
+## Connect to database
+
+You can connect to and interact with an SQL database using specific functions from the [DataFrame library](https://kotlin.github.io/dataframe/gettingstarted.html). 
+You can use `DatabaseConfiguration` to establish a connection to your database and `getSchemaForAllSqlTables` to retrieve 
+the schema of all tables within it.
+
+Let's look at an example:
+
+1. **Import libraries:** Import the DataFrame library, which is essential for data manipulation tasks.
+This is done by running the following command in a code cell:
+
+```kotlin
+%use dataframe
+```
+
+2. **Establish database connection:** Use `DatabaseConfiguration` to define your database's connection parameters, 
+including the URL, username, and password:
+
+```kotlin
+val URL = "YOUR_URL"
+val USER_NAME = "YOUR_USERNAME"
+val PASSWORD = "YOUR_PASSWORD"
+
+val dbConfig = DatabaseConfiguration(URL, USER_NAME, PASSWORD)
+```
+
+3. **Retrieve table schemas:** Once connected, use the `getSchemaForAllSqlTables()` function to fetch and display the 
+schema information for each table in the database:
+
+```kotlin
+val dataschemas = DataFrame.getSchemaForAllSqlTables(dbConfig)
+
+dataschemas.forEach { 
+    println("---Yet another table schema---")
+    println(it)
+    println()
+}
+```
+
+For more information on connecting to SQL databases, see [Read from SQL databases in the DataFrame documentation](https://kotlin.github.io/dataframe/readsqldatabases.html).
+
+## Retrieve and manipulate data
+
+After [establishing a connection to an SQL database](#connect-to-database), you can retrieve and manipulate data in Kotlin Notebook, utilizing the DataFrame library. 
+You can use `readAllSqlTables` to retrieve data and methods, such as [`.filter`](https://kotlin.github.io/dataframe/filter.html), [`.groupBy`](https://kotlin.github.io/dataframe/groupby.html), 
+and [`.convert`](https://kotlin.github.io/dataframe/convert.html) to manipulate data. 
+
+Let's look at an example of connecting to an IMDB database and retrieving data about movies directed by Quentin Tarantino:
+
+1. **Fetch data from all tables:** Retrieve data from all tables using the `readAllSqlTables` function:
```

Comment:
Sorry, could you please concentrate more on the function for reading data from one sql table instead of reading all (it's experimental functuionality and could change in next release)

Reply:
Sure, I'll change it to an example with `readSqlTable` 👍